canvas/src/layer/layer_2d.rs

```diff
@@ -3,32 +3,161 @@ use crate::{
 };
 pub use glyph_brush::{GlyphCruncher, HorizontalAlign, Layout, VerticalAlign};
 pub use lyon::{
-  path::{builder::PathBuilder, traits::PathIterator, Path, Winding},
+  geom::Arc,
+  math::point,
+  path::{builder::PathBuilder, path::Builder, traits::PathIterator, EndpointId, Path, Winding},
   tessellation::*,
 };
+
 use std::{
   cmp::PartialEq,
   ops::{Deref, DerefMut},
 };
 
+pub type Vector = euclid::default::Vector2D<f32>;
+pub type Angle = euclid::Angle<f32>;
+
 /// The 2d layer is a two-dimensional grid. The coordinate (0, 0) is at the
 /// upper-left corner of the canvas. Along the X-axis, values increase towards
 /// the right edge of the canvas. Along the Y-axis, values increase towards the
 /// bottom edge of the canvas.
-#[derive(Default, Debug, Clone)]
+// #[derive(Default, Debug, Clone)]
 pub struct Rendering2DLayer<'a> {
   state_stack: Vec<State>,
   pub(crate) commands: Vec<Command<'a>>,
+  path: Option<Builder>,
 }
 
 impl<'a> Rendering2DLayer<'a> {
   pub fn new() -> Self {
     Self {
       state_stack: vec![State::new()],
       commands: vec![],
+      path: None,
     }
   }
 
+  #[inline]
+  pub fn get_path(&mut self) -> Option<Path> { self.path.take().map(|b| b.build()) }
+
+  /// Starts a new path by emptying the list of sub-paths.
+  /// Call this method when you want to create a new path.
+  #[inline]
+  pub fn begin_path(&mut self, x: f32, y: f32) -> &mut Self {
+    self.path = Some(Builder::new());
```

Comment:
A private tiny method of `self.path.get_or_insert(|| Builder::new())` will very useful, in these methods.